Update Developer Guide re. tab completion (#9918)

JulienCochuyt opened the pull request:
<!--
Please fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples
-->

### Link to issue number:

Fixes #9918

### Summary of the issue:

In the Python console, the tab-completion is smarter with Python 3.

### Description of how this pull request fixes the issue:

Document the new behavior in the Developer Guide.

### Testing performed:

None.

### Known issues with pull request:

### Change log entry:

Section: Changes for developpers

The tab-completion in the Python console is now smarter. Refer to the Developer Guide for more details.

JulienCochuyt committed:
Update Developer Guide re. tab completion (#9918)

XLTechie commented on developerGuide.t2t:
```diff
@@ -796,6 +796,15 @@ These variables are:
+Hit the tab key once to complete the current input if there is one single candidate.
+If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
+By default, only "public" member attributes are listed.
+That is, if the input is "nav.", attributes names with no leading underscore are proposed.
```
I think that should be "attribute names"

XLTechie replied:
```suggestion
That is, if the input is "nav.", attribute names with no leading underscore are proposed.
```

XLTechie commented on developerGuide.t2t:
```diff
@@ -796,6 +796,15 @@ These variables are:
+By default, only "public" member attributes are listed.
+That is, if the input is "nav.", attributes names with no leading underscore are proposed.
+If the input is "nav._", attributes names with a single leading underscore are proposed.
+Similarily, if the input is "nav.__", attributes names with two leading underscores are proposed.
```
S/b "Similarly"

XLTechie replied:
All three of those s/b "attribute names" IMO.

```suggestion
That is, if the input is "nav.", attributes names with no leading underscore are proposed.
If the input is "nav._", attributes names with a single leading underscore are proposed.
Similarily, if the input is "nav.__", attributes names with two leading underscores are proposed.
```
```suggestion
That is, if the input is "nav.", attribute names with no leading underscore are proposed.
If the input is "nav._", attribute names with a single leading underscore are proposed.
Similarly, if the input is "nav.__", attribute names with two leading underscores are proposed.
```

XLTechie commented:
Apologies, that suggested change did not work as intended to include multiple 
lines in the diff.

I will delete it if it lets me.

The idea was to suggest changing all three "attributes names" to "attribute 
names", and "similarily" to "similarly".

JulienCochuyt committed:
Fix-up

 - https://github.com/nvaccess/nvda/pull/9935#discussion_r303851784
 - https://github.com/nvaccess/nvda/pull/9935#discussion_r303855384

XLTechie commented on developerGuide.t2t:
```diff
@@ -796,6 +796,15 @@ These variables are:
+If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
+By default, only "public" member attributes are listed.
+That is, if the input is "nav.", attributes names with no leading underscore are proposed.
+If the input is "nav._", attributes names with a single leading underscore are proposed.
```
```suggestion
If the input is "nav._", attribute names with a single leading underscore are proposed.
```

JulienCochuyt commented:
@XLTechie, thank you for the feedback: I'm not a native English speaker.

XLTechie commented:
Sorry for the overlapping inline comments. Your "fix-up" commit wasn't showing 
when I posted them, but the notification for it came in before the notifications 
for my corrected inlines.
Not sure what the deal was there, but sorry about all the confusion.

XLTechie commented on developerGuide.t2t:
```diff
@@ -801,9 +801,9 @@ The input control supports tab-completion of variables and member attributes nam
 Hit the tab key once to complete the current input if there is one single candidate.
 If there are more than one, hit the tab key a second time to open a menu listing all matching possibilities.
```
I Missed this one, although it is not really important.

```suggestion
If there is more than one, hit the tab key a second time to open a menu listing all matching possibilities.
```

JulienCochuyt committed:
Fix-up (https://github.com/nvaccess/nvda/pull/9935#discussion_r303890023)

Co-Authored-By: Luke Davis <8139760+XLTechie@users.noreply.github.com>

JulienCochuyt commented:
> Sorry for the overlapping inline comments

No problem, feel free to further improve the proposed phrasing.

XLTechie commented:
I don't see anything else to improve.:) The only important one was 
"similarly"--all the rest were just me being a pedantic grammarian.

feerrenrut reviewed (approved):
Thanks @JulienCochuyt
